apiserver/authentication/jwt/jwt.go

```diff
@@ -22,17 +19,32 @@ import (
 	"github.com/juju/juju/core/permission"
 )
 
+// Authenticator defines an interface for authenticating requests.
 type Authenticator interface {
 	authentication.RequestAuthenticator
-	TokenParser
+}
+
+// TokenParser defines an interface for parsing JWT tokens.
+type TokenParser interface {
+	// Parse accepts a base64 string and extracts a JWT token.
+	// This method should return a NotProvisioned error if
+	// the parser is not ready.
+	Parse(ctx context.Context, tok string) (jwt.Token, error)
 }
 
 // JWTAuthenticator is an authenticator responsible for handling JWT tokens from
 // a client.
 type JWTAuthenticator struct {
-	cache      *jwk.Cache
-	httpClient *http.Client
-	refreshURL string
+	parser TokenParser
+}
+
+// NewAuthenticator creates a new JWT authenticator with the supplied parser getter.
```

Comment:
looking at the L44, we supply the token parser, not a parser getter.

Reply:
Fixed, thanks.